coriolis/osmorphing/windows.py

```diff
@@ -433,8 +467,89 @@ def _install_cloudbase_init(self, download_url,
+                subnet_mask = self._conn.exec_ps_command(
+                    "(Get-ItemProperty -Path '%s').SubnetMask" % reg_path)
+                if not (ip_address and subnet_mask):
+                    LOG.warning(
+                        "No IP Address or Subnet Mask found for interface: "
+                        "'%s'" % interface)
+                    continue
+                prefix_length = ipaddress.IPv4Network(
+                    (0, subnet_mask)).prefixlen
+                default_gateway = self._conn.exec_ps_command(
+                    "(Get-ItemProperty -Path '%s').DefaultGateway" % reg_path)
+                name_server = self._conn.exec_ps_command(
+                    "(Get-ItemProperty -Path '%s').NameServer" % reg_path)
+                ip_info = {
+                    "ip_address": ip_address,
+                    "prefix_length": prefix_length,
+                    "default_gateway": default_gateway,
+                    "dns_addresses": name_server}
+                LOG.debug(
+                    "Found static IP configuration for interface '%s': "
+                    "%s" % (interface, ip_info))
+                ips_info.append(ip_info)
+            else:
+                LOG.debug(
+                    "Could not find a static IP configuration for interface: "
+                    "'%s'" % interface)
+                continue
+
+        return ips_info
+
+    def _check_ips_info(self, nics_info, ips_info):
+        source_ip_addresses = set()
+        reg_ip_addresses = set()
+        for nic in nics_info:
+            source_ip_addresses.update(nic['ip_addresses'])
+
+        for ip_info in ips_info:
+            reg_ip_addresses.add(ip_info['ip_address'])
+
+        diff = source_ip_addresses - reg_ip_addresses
+        if diff:
+            raise exception.OSMorphingException(
+                "The following IP(s): %s found on the source VM's NICs were "
+                "not found in the registry. Please check whether the static "
+                "IP configurations on the source machine are properly set up "
+                "and retry the migration." % diff)
+
+    def _write_static_ip_script(self, base_dir, nics_info, ips_info):
+        scripts_dir = self._get_cbslinit_scripts_dir(base_dir)
+        script_path = "%s\\01-static-ip-config.ps1" % scripts_dir
+        contents = STATIC_IP_SCRIPT_TEMPLATE % {'nics_info': nics_info,
+                                                'ips_info': ips_info}
+        utils.write_winrm_file(self._conn, script_path, contents.encode())
+
     def set_net_config(self, nics_info, dhcp):
-        pass
+        if dhcp:
+            return
+
+        key_name = str(uuid.uuid4())
+        self._load_registry_hive(
+            "HKLM\\%s" % key_name,
+            "%sWindows\\System32\\config\\SYSTEM" % self._os_root_dir)
+        try:
+            cbslinit_base_dir = "%sCloudbase-Init" % self._os_root_dir
```

Comment:
Please abstract this as a method/constant as it's used through the rest of this module too and we should avoid "desyncs".